locobot/agent/locobot_mover.py

```diff
@@ -38,16 +45,17 @@ def __init__(self, ip=None, backend="locobot"):
         self.close_loop = False if backend == "habitat" else True
         self.curr_look_dir = np.array([0, 0, 1])  # initial look dir is along the z-axis
 
-        intrinsic_mat = self.bot.get_intrinsics()
+        intrinsic_mat = safe_call(self.bot.get_intrinsics)
```

Comment:
i'm not convinced that these `safe_call`s are actually future-safe.

Presumably, the Pyro4 connection can be closed at any point in time, not just in these specific `safe_call`s in this PR.
So,
1. Why did you only make these 3 callsites to be `safe_call` and leave the others as-is?
2. What happens if a `safe_call` fails while doing something critical or important? Should we actually have a `reset() / revert()` mechanism for the robot to undo or get out of a multi-step task that it gave up on?
3. Why not do a `retry_call` where it retries a few times with a `time.sleep` in between, and if none of it works, then actually just fatally crap out?

Reply:
> i'm not convinced that these `safe_call`s are actually future-safe.
> 
> Presumably, the Pyro4 connection can be closed at any point in time, not just in these specific `safe_call`s in this PR.
> So,
> 
> 1. Why did you only make these 3 callsites to be `safe_call` and leave the others as-is?
> 2. What happens if a `safe_call` fails while doing something critical or important? Should we actually have a `reset() / revert()` mechanism for the robot to undo or get out of a multi-step task that it gave up on?
> 3. Why not do a `retry_call` where it retries a few times with a `time.sleep` in between, and if none of it works, then actually just fatally crap out?

Answers:
1. Can easily add `safe_call` to all callsites. Added this only to the constructor and `get_base_state` because these are the canaries in the coal mine in a way - they get called the first and the most times. Also, anecdotally haven't run into exceptions with the other APIs and I don't know of a reliable way to simulate their errors either (and hence don't know everything we'll need to (or not) catch).
2. Some of the Pyro4 errors succeed on retries. If they continue to fail fatally, the exception will be raised eventually based on the mechanism here https://github.com/facebookresearch/droidlet/blob/bf813047ed0e74479420217d47f8b990e43d5231/base_agent/loco_mc_agent.py#L163.
3. We can do that, but the design currently essentially achieves that (ie maybe succeeds on retried or fails fatally eventually). Don't think it is worth the investment now given that these errors are only sporadic.

Reply:
Also, good call out about `reset() / revert()` and this does not fix that. Let's cross that bridge when we get there?

Reply:
on (2), the important distinction is whether they are retried before the current codepath / task is finished, or they are retried on the next execution.

"bot go to the door"

`safe_call` fails, and the bot to the door task fails.

"bot go to the door"

`retry_call` tries ~5 times with sleep(1), and passes, continues and finishes the task.

That's the important distinction / decision to deliberately make, whether the call will retry before giving up on the current codepath and raising an exception.

Reply:
That makes sense. Added retries with wait using https://github.com/jd/tenacity